### ISSUE TITLE
Agent factory sometimes falls over immediately when built against libstdc++

### ISSUE THREAD

**Issue by SwooshyCueb** [irods/irods#7745]
When iRODS is built against libstdc++, sometimes (about half the time, I think, assuming some other problem doesn't occur first) the agent factory falls over immediately, causing the server to exit.

Log:
```
 {"log_category":"server","log_level":"info","log_message":"Initializing server ...","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.530Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"server","log_level":"info","log_message":"Setting up UNIX domain socket for agent factory ...","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.540Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"server","log_level":"info","log_message":"Forking agent factory ...","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.544Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"server","log_level":"info","log_message":"Connecting to agent factory [agent_factory_pid=697] ...","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.545Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"agent_factory","log_level":"info","log_message":"Initializing agent factory ...","server_host":"93c23b681ce5","server_pid":697,"server_timestamp":"2024-05-15T19:14:46.549Z","server_type":"agent_factory","server_zone":"tempZone"}
 {"log_category":"server","log_level":"debug","log_message":"Setting stacktrace dump signal handler for process [697].","server_host":"93c23b681ce5","server_pid":697,"server_timestamp":"2024-05-15T19:14:46.549Z","server_type":"agent_factory","server_zone":"tempZone"}
 {"log_category":"server","log_level":"debug","log_message":"Stacktraces will be dumped to [/var/lib/irods/stacktraces].","server_host":"93c23b681ce5","server_pid":697,"server_timestamp":"2024-05-15T19:14:46.549Z","server_type":"agent_factory","server_zone":"tempZone"}
 {"local_hostname":"localhost, 93c23b681ce5","log_category":"server","log_level":"info","port":"1247","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.608Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"server","log_level":"info","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.645Z","server_type":"server","server_zone":"tempZone","zone_info.host":"localhost","zone_info.name":"tempZone","zone_info.port":"1247","zone_info.type":"LOCAL_ICAT"}
 {"log_category":"server","log_level":"info","log_message":"rodsServer Release version rods4.90.0 - API Version d is up","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.646Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"legacy","log_level":"info","log_message":">>> control plane :: listening on port 1248\n","server_host":"93c23b681ce5","server_pid":696,"server_timestamp":"2024-05-15T19:14:46.649Z","server_type":"server","server_zone":"tempZone"}
 {"log_category":"agent_factory","log_level":"info","log_message":"The rodsServer socket peer has shut down.","server_host":"93c23b681ce5","server_pid":697,"server_timestamp":"2024-05-15T19:14:46.875Z","server_type":"agent_factory","server_zone":"tempZone"}
 {"log_category":"server","log_level":"critical","log_message":"Agent factory returned with error code [0].","server_host":"93c23b681ce5","server_pid":697,"server_timestamp":"2024-05-15T19:14:46.875Z","server_type":"agent_factory","server_zone":"tempZone"}
```

This problem is hampering my ability to test the main branch of irods with the new externals, so I'm going to try and fix it as part of my work towards Ubuntu 24.04 support.

**Comment by SwooshyCueb** [comment]
I tried switching all usage of `boost::unordered_map` to `std::unordered_map` to fix #7747. It didn't really work, but it did *drastically* reduce the frequency at which this problem occurs. This took me by surprise as I didn't think the two issues to be related.
I have not observed this issue in my runner container since applying the change, but I have observed it still occurring in the testing environment.
